### PR TITLE
Implement unit conversion for a wider range of units

### PR DESCRIPTION
**Pull request by connortann**
Adds a pint unit registry, that can be used to convert between a wide range of units.

Eventually, should be able to handle a large set of units, including all valid resqml uoms.

### CODE REVIEW

File: resqpy/olio/data/unit_registry.txt [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    square_yard = yard ** 2 = sq_yd
+    square_mile = mile ** 2 = sq_mi
+
+    cubic_inch = in ** 3 = cu_in
+    cubic_foot = ft ** 3 = cu_ft = cubic_feet
+    cubic_yard = yd ** 3 = cu_yd
+@end
+
+@group USCSLengthSurvey
+    survey_foot = 1200 / 3937 * meter = sft = ft[US]
```
Comment (andy-beer) [inline resqpy/olio/data/unit_registry.txt:97]: Have we checked that a 'survey foot' is actually the same as 'ft[US]'?

File: tests/test_units.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        bwam.parse_unit('foobar')
+
+
+def test_strip_scaling_prefix():
+    assert bwam.strip_scaling_prefix("ft") == (1, "ft")
+    assert bwam.strip_scaling_prefix("100 ft") == (100, "ft")
+    assert bwam.strip_scaling_prefix("1/32 ft") == (1/32, "ft")
+    assert bwam.strip_scaling_prefix("1e9 ft") == (1e9, "ft")
+    assert bwam.strip_scaling_prefix("1.2 ft") == (1.2, "ft")
+
```
Comment (andy-beer) [inline tests/test_units.py:45]: perhaps add tests for prefixes of form: '0.001' and '1E6'

**Review by andy-beer — COMMENTED** [review]
Let's talk it over again – I still have mixed feelings!

**Comment by connortann** [comment]
Thanks for the discussion Andy! I'll close this PR for now without merging. To recap our current plan:

1. Implementing RESQML units of measure explicitly from a built JSON database looks more promising than using pint
2. We could repurpose some of this code in other repositories that would benefit from broader, general-purpose unit conversion
3. We can cherry-pick the tests from this branch and use them with our future alternate implementation

**Pull request closed without merging**